libp2p/P2PSession.cpp

```diff
@@ -163,10 +171,27 @@ void P2PSession::onTopicMessage(P2PMessage::Ptr message)
                                         }
                                         else
                                         {
-                                            topicList->insert(topics[i]);
+                                            if (versionLe2)
```

Comment:
Can we split the large function into small pieces?
It is too large to maintain now.
More, please encapsulate your new code instead of appending or inserting rudely.

Reply:
done